internal/factsengine/factsengine.go

```diff
@@ -133,7 +136,29 @@ func (c *FactsEngine) Listen(ctx context.Context) error {
 }
 
 func (c *FactsEngine) handleRequest(contentType string, request []byte) error {
-	factsRequests, err := parseFactsRequest(request)
+	event, err := contracts.NewFactsGatheringRequestedV1FromJsonCloudEvent(request)
+
+	if err != nil {
+		return err
+	}
+
+	agentID := agentUuid
+
+	var interested bool = false
+	for _, target := range event.Targets {
+		if target == agentID {
```

Comment:
this determines whether the running agent should process or not the consumed message. 
Sort of _am I among the targets of current execution_?